packages/firebase-auth/vitest.config.ts

```diff
@@ -1,7 +1,29 @@
+import type { Plugin } from 'vitest/config'
 import { defineProject } from 'vitest/config'
 
+const firebasePlugin = {
+  name: 'firebase',
+  async configureServer(server) {
+    const { default: client } = await import('firebase-tools')
+
+    void client.emulators.start({
+      cwd: server.config.root,
+      nonInteractive: true,
+      project: 'example-project12345',
+      projectDir: server.config.root,
+    })
+  },
+  async buildEnd() {
+    const { default: controller } = await import('firebase-tools/lib/emulator/controller')
+
+    await controller.cleanShutdown()
+  },
+} satisfies Plugin
+
 export default defineProject({
   test: {
     globals: true,
   },
+
+  plugins: [firebasePlugin],
```

Comment:
Instead of running the `firebase-auth` tests with `firebase emulators:exec`, I put together a plugin that starts/stops the emulator using the Vite dev server

This means tests can be run with the regular `vitest` command. It also enables collecting code coverage